pkg/initializer/validator/datavalidator.go

```diff
@@ -88,6 +89,35 @@ func (d *DataValidator) Validate(mode Mode, failBelowRevision int64) (DataDirSta
 }
 
 func (d *DataValidator) sanityCheck(failBelowRevision int64) (DataDirStatus, error) {
+	mntDataDir := path.Dir(d.Config.DataDir)
+	path := mntDataDir + "/" + safeGuard
+	namespace := os.Getenv(podNamespace)
+	if namespace == "" {
+		d.Logger.Warn("POD_NAMESPACE environment variable is not set. The variable is used to safe guard against wrong volume mount")
```

Comment:
Should we change this to `d.Logger.Fatal`?
The volume would still be mounted if the `POD_NAMESPACE` env var is not present

wdyt?

Reply:
We shouldn't be changing it to Fatal. This warning is raised in case of unit tests. In Unit test or if you run ETCD BR standalone, you may not provide the Env variable, so a warning is raised